pkg/framework/proxies.go

```diff
@@ -0,0 +1,196 @@
+package framework
+
+import (
+	"context"
+	"errors"
+	configv1 "github.com/openshift/api/config/v1"
+	kappsapi "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func DeployClusterProxy(c runtimeclient.Client) (kappsapi.Deployment, error) {
+	configMap := v1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "squid-conf",
+			Namespace: "default",
+		},
+		Data: map[string]string{
+			"squid.conf": "acl localnet src 192.168.0.0/16\t# RFC1918 possible internal network\nacl crcnet src 10.88.0.0/16\t# RFC1918 possible internal network\nacl SSL_ports port 443\nacl Safe_ports port 80\t\t# http\nacl Safe_ports port 21\t\t# ftp\nacl Safe_ports port 443\t\t# https\nacl Safe_ports port 70\t\t# gopher\nacl Safe_ports port 210\t\t# wais\nacl Safe_ports port 1025-65535\t# unregistered ports\nacl Safe_ports port 280\t\t# http-mgmt\nacl Safe_ports port 488\t\t# gss-http\nacl Safe_ports port 591\t\t# filemaker\nacl Safe_ports port 777\t\t# multiling http\nacl CONNECT method CONNECT\nhttp_access deny !Safe_ports\nhttp_access deny CONNECT !SSL_ports\nhttp_access allow localhost manager\nhttp_access deny manager\nacl mysubnet src 192.168.0.0/16\nhttp_access allow localhost localnet crcnet\nhttp_access allow all\nhttp_port 3128\ncoredump_dir /var/spool/squid\nrefresh_pattern ^ftp:\t\t1440\t20%\t10080\nrefresh_pattern ^gopher:\t1440\t0%\t1440\nrefresh_pattern -i (/cgi-bin/|\\?) 0\t0%\t0\nrefresh_pattern (Release|Packages(.gz)*)$      0       20%     2880\nrefresh_pattern .\t\t0\t20%\t4320",
```

Comment:
This config should be moved to a constant, ideally a multiline constant so that we don't have to have all of the escape characters in it, can you update please?

Reply:
sure thing